File: backend/src/main/java/github/lms/lemuel/xr/common/chatops/TelegramChatOpsProperties.java
package github.lms.lemuel.xr.common.chatops;

import org.springframework.boot.context.properties.ConfigurationProperties;

/** chatops.telegram.* 설정. */
@ConfigurationProperties(prefix = "chatops.telegram")
public record TelegramChatOpsProperties(
        boolean enabled,
        String botToken,
        String chatId,
        String baseUrl     // 기본 https://api.telegram.org
) {
    public TelegramChatOpsProperties {
        if (baseUrl == null || baseUrl.isBlank()) baseUrl = "https://api.telegram.org";
    }
}

File: backend/src/main/kotlin/github/lms/lemuel/xr/common/chatops/TelegramChatOpsProperties.kt
package github.lms.lemuel.xr.common.chatops

import org.springframework.boot.context.properties.ConfigurationProperties

/** chatops.telegram.* 설정. */
@ConfigurationProperties(prefix = "chatops.telegram")
class TelegramChatOpsProperties(
    val enabled: Boolean,
    val botToken: String?,
    val chatId: String?,
    baseUrl: String?,     // 기본 https://api.telegram.org
) {
    val baseUrl: String = if (baseUrl.isNullOrBlank()) "https://api.telegram.org" else baseUrl
}
